iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/AbstractIdentifiable.java

```diff
@@ -58,17 +60,43 @@ public String getNameOrId() {
         return Collections.unmodifiableSet(aliases);
     }
 
+    @Override
+    public Set<String> getAliases(String aliasType) {
```

Comment:
This is a very inefficient implementation. It depends how it is used but we have to be aware of potential performance issue using this method. Is it really needed?

Reply:
I made a draft PR #1447 to show aliases' types can be used in CGMES update and you're right: we don't really need this method (or at least not in this form). Types are actually unique for one identifiable if they are not null in this use case (in the example, a switch can only have one alias associated with a "terminal1" type). I need to rethink how to do this.